src/solvers/prop/bdd_expr.h

```diff
@@ -36,6 +37,110 @@ class bdd_exprt
   bddt from_expr(const exprt &expr);
   exprt as_expr(const bddt &root) const;
 
+  template <typename returnt>
+  struct strategyt
+  {
+    std::function<returnt(const exprt &)> leaf;
+    std::function<returnt(const returnt &, const returnt &)> make_or;
+    std::function<returnt(const returnt &, const returnt &)> make_and;
+    std::function<returnt(const returnt &)> make_not;
+    std::function<returnt(const returnt &, const returnt &, const returnt)> make_ite;
```

Comment:
I don't understand `ite`. After a minute I guessed "if then else". Please make it clearer somehow.